File: src/t905_SortArrayByParity.java
public class t905_SortArrayByParity {
    public int[] sortArrayByParity(int[] arr) {
        int i = 0;
        int j = arr.length - 1;
        int temp = 0;
        while (i < j) {
            if (arr[i] % 2 == 0) {
                i++;
                continue;
            }
            if (arr[i] % 2 == 1) {
                j--;
                continue;
            }
            temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            i++;
            j--;
        }
        return arr;
    }
}

File: src/python/t905_SortArrayByParity.py
def sortArrayByParity(self, nums: List[int]) -> List[int]:
    i = 0
    j = len(nums) - 1
    temp = 0
    while i < j:
        if nums[i] % 2 == 0:
            i = i + 1
            continue
        if nums[j] % 2 == 1:
            j = j - 1
            continue
        temp = nums[i]
        nums[i] = nums[j]
        nums[j] = temp
        i = i + 1
        j = j - 1
    return nums